src/bin/inx-chronicle/api/error.rs

```diff
@@ -4,140 +4,205 @@
-        ApiError::BadParse(err.into())
+impl IntoResponse for UnimplementedError {
+    fn into_response(self) -> axum::response::Response {
+        ApiError::from(self).into_response()
     }
 }
 
-impl<T: Into<InternalApiError>> From<T> for ApiError {
-    fn from(err: T) -> Self {
-        ApiError::Internal(err.into())
+#[derive(Error, Debug)]
+#[allow(missing_docs)]
+pub enum MissingError {
+    #[error("no results returned")]
+    NoResults,
+    #[error("no endpoint found")]
+    NotFound,
+}
+
+impl ErrorStatus for MissingError {
+    fn status(&self) -> StatusCode {
+        StatusCode::NOT_FOUND
     }
 }
 
-impl IntoResponse for ApiError {
+impl IntoResponse for MissingError {
     fn into_response(self) -> axum::response::Response {
-        ErrorBody::from(self).into_response()
+        ApiError::from(self).into_response()
     }
 }
 
 #[derive(Error, Debug)]
-pub enum ParseError {
-    #[allow(dead_code)]
-    #[error("Invalid cursor")]
+pub enum RequestError {
+    #[error("invalid cursor")]
     BadPagingState,
+    #[error("invalid time range")]
+    BadTimeRange,
     #[cfg(feature = "stardust")]
-    #[error(transparent)]
-    BeeBlockStardust(#[from] iota_types::block::Error),
-    #[error(transparent)]
+    #[error("invalid IOTA Stardust data: {0}")]
+    IotaStardust(#[from] iota_types::block::Error),
+    #[error("invalid bool value provided: {0}")]
     Bool(#[from] ParseBoolError),
-    #[error(transparent)]
+    #[error("invalid U256 value provided: {0}")]
     DecimalU256(#[from] uint::FromDecStrErr),
-    #[error(transparent)]
+    #[error("invalid integer value provided: {0}")]
     Int(#[from] ParseIntError),
-    #[error(transparent)]
+    #[error("invalid authorization header provided: {0}")]
+    InvalidAuthHeader(#[from] TypedHeaderRejection),
+    #[error("invalid query parameters provided: {0}")]
+    InvalidQueryParams(#[from] QueryRejection),
+    #[error("invalid sort order provided: {0}")]
     SortOrder(#[from] ParseSortError),
-    #[error(transparent)]
-    TimeRange(#[from] time::error::ComponentRange),
+}
+
+impl ErrorStatus for RequestError {
+    fn status(&self) -> StatusCode {
+        StatusCode::BAD_REQUEST
+    }
 }
 
 #[derive(Error, Debug)]
 pub enum ConfigError {
-    #[error(transparent)]
+    #[error("invalid allow-origin header in config: {0}")]
     InvalidHeader(#[from] InvalidHeaderValue),
-    #[error(transparent)]
+    #[error("invalid hex value in config: {0}")]
     InvalidHex(#[from] hex::FromHexError),
-    #[error("Invalid regex in config: {0}")]
+    #[error("invalid regex in config: {0}")]
     InvalidRegex(#[from] regex::Error),
-    #[error(transparent)]
+    #[error("invalid secret key: {0}")]
     SecretKey(#[from] super::secret_key::SecretKeyError),
-    #[error(transparent)]
-    TimeConversion(#[from] time::error::ConversionRange),
 }
+
+impl ErrorStatus for ConfigError {
+    fn status(&self) -> StatusCode {
+        StatusCode::BAD_REQUEST
```

Comment:
Are we sure about this? Shouldn't that be a `5**`? :thinking:

Reply:
Hmm...perhaps.